Fix unused typedef warnings

These all pop when building the release version with clang.

```c++
<path_to>/deal.II/source/fe/fe_poly_tensor.cc:174:42: warning: unused typedef 'FEE' [-Wunused-local-typedef]
  typedef    FiniteElement<dim,spacedim> FEE;
                                         ^
<path_to>/deal.II/source/fe/fe_poly_tensor.cc:219:42: warning: unused typedef 'FEE' [-Wunused-local-typedef]
  typedef    FiniteElement<dim,spacedim> FEE;
                                         ^
<path_to>/deal.II/source/fe/fe_poly_tensor.cc:266:42: warning: unused typedef 'FEE' [-Wunused-local-typedef]
  typedef    FiniteElement<dim,spacedim> FEE;
                                         ^
```
and
```c++
<path_to>/deal.II/source/fe/fe_system.cc:2175:39: warning: unused typedef 'FEL' [-Wunused-local-typedef]
  typedef FiniteElement<dim,spacedim> FEL;
                                      ^
<path_to>/deal.II/source/fe/fe_system.cc:2198:40: warning: unused typedef 'FEL' [-Wunused-local-typedef]
  typedef  FiniteElement<dim,spacedim> FEL;
                                       ^
```
and
```c++
<path_to>/deal.II/include/deal.II/fe/fe_tools.templates.h:1455:41: warning: unused typedef 'FEL' [-Wunused-local-typedef]
    typedef FiniteElement<dim,spacedim> FEL;
                                        ^
```

--- a/include/deal.II/fe/fe_tools.templates.h
+++ b/include/deal.II/fe/fe_tools.templates.h
@@ -1452,9 +1452,9 @@ namespace FETools
     const std::vector<Point<dim> > &
     fe2_support_points = fe2.get_unit_support_points ();
 
-    typedef FiniteElement<dim,spacedim> FEL;
     Assert(fe2_support_points.size()==fe2.dofs_per_cell,
-           typename FEL::ExcFEHasNoSupportPoints());
+           (typename FiniteElement<dim,spacedim>::
+            ExcFEHasNoSupportPoints()));
 
     for (unsigned int i=0; i<fe2.dofs_per_cell; ++i)
       {

--- a/include/deal.II/numerics/vector_tools.templates.h
+++ b/include/deal.II/numerics/vector_tools.templates.h
@@ -4522,10 +4522,8 @@ namespace VectorTools
                   // cannot check this.
                   if (dynamic_cast<const FESystem<dim>*> (&cell->get_fe ()) == nullptr)
                     {
-                      typedef FiniteElement<dim> FEL;
                       AssertThrow (dynamic_cast<const FE_Nedelec<dim>*> (&cell->get_fe ()) != nullptr,
-
-                                   typename FEL::ExcInterpolationNotImplemented ());
+                                   (typename FiniteElement<dim>::ExcInterpolationNotImplemented ()));
                     }
 
                   for (unsigned int dof = 0; dof < dofs_per_face; ++dof)
@@ -4606,10 +4604,8 @@ namespace VectorTools
                   // check this.
                   if (dynamic_cast<const FESystem<dim>*> (&cell->get_fe ()) == nullptr)
                     {
-                      typedef FiniteElement<dim> FEL;
-
                       AssertThrow (dynamic_cast<const FE_Nedelec<dim>*> (&cell->get_fe ()) != nullptr,
-                                   typename FEL::ExcInterpolationNotImplemented ());
+                                   typename FiniteElement<dim>::ExcInterpolationNotImplemented ());
                     }
 
                   for (unsigned int dof = 0; dof < dofs_per_face; ++dof)
@@ -4721,10 +4717,8 @@ namespace VectorTools
                   // element. If the FE is a FESystem we cannot check this.
                   if (dynamic_cast<const FESystem<dim>*> (&cell->get_fe ()) == nullptr)
                     {
-                      typedef FiniteElement<dim> FEL;
-
                       AssertThrow (dynamic_cast<const FE_Nedelec<dim>*> (&cell->get_fe ()) != nullptr,
-                                   typename FEL::ExcInterpolationNotImplemented ());
+                                   typename FiniteElement<dim>::ExcInterpolationNotImplemented ());
                     }
 
                   const unsigned int dofs_per_face = cell->get_fe ().dofs_per_face;
@@ -4798,10 +4792,8 @@ namespace VectorTools
                   // element. If the FE is a FESystem we cannot check this.
                   if (dynamic_cast<const FESystem<dim>*> (&cell->get_fe ()) == nullptr)
                     {
-                      typedef FiniteElement<dim> FEL;
-
                       AssertThrow (dynamic_cast<const FE_Nedelec<dim>*> (&cell->get_fe ()) != nullptr,
-                                   typename FEL::ExcInterpolationNotImplemented ());
+                                   typename FiniteElement<dim>::ExcInterpolationNotImplemented ());
                     }
 
                   const unsigned int superdegree = cell->get_fe ().degree;
@@ -5549,9 +5541,8 @@ namespace VectorTools
                           // If the FE is a FESystem we cannot check this.
                           if (dynamic_cast<const FESystem<dim>*> (&cell->get_fe ()) == nullptr)
                             {
-                              typedef FiniteElement<dim> FEL;
                               AssertThrow (dynamic_cast<const FE_Nedelec<dim>*> (&cell->get_fe ()) != nullptr,
-                                           typename FEL::ExcInterpolationNotImplemented ());
+                                           typename FiniteElement<dim>::ExcInterpolationNotImplemented ());
 
                             }
 
@@ -5646,10 +5637,8 @@ namespace VectorTools
                           // If the FE is a FESystem we cannot check this.
                           if (dynamic_cast<const FESystem<dim>*> (&cell->get_fe ()) == nullptr)
                             {
-                              typedef FiniteElement<dim> FEL;
-
                               AssertThrow (dynamic_cast<const FE_Nedelec<dim>*> (&cell->get_fe ()) != nullptr,
-                                           typename FEL::ExcInterpolationNotImplemented ());
+                                           typename FiniteElement<dim>::ExcInterpolationNotImplemented ());
                             }
 
                           const unsigned int superdegree = cell->get_fe ().degree;
@@ -6011,10 +6000,8 @@ namespace VectorTools
                   // check this.
                   if (dynamic_cast<const FESystem<dim>*> (&cell->get_fe ()) == nullptr)
                     {
-                      typedef FiniteElement<dim> FEL;
-
                       AssertThrow (dynamic_cast<const FE_RaviartThomas<dim>*> (&cell->get_fe ()) != nullptr,
-                                   typename FEL::ExcInterpolationNotImplemented ());
+                                   typename FiniteElement<dim>::ExcInterpolationNotImplemented ());
                     }
 
                   fe_values.reinit (cell, face + cell->active_fe_index ()
@@ -6075,10 +6062,8 @@ namespace VectorTools
                   // check this.
                   if (dynamic_cast<const FESystem<dim>*> (&cell->get_fe ()) == nullptr)
                     {
-                      typedef FiniteElement<dim> FEL;
-
                       AssertThrow (dynamic_cast<const FE_RaviartThomas<dim>*> (&cell->get_fe ()) != nullptr,
-                                   typename FEL::ExcInterpolationNotImplemented ());
+                                   typename FiniteElement<dim>::ExcInterpolationNotImplemented ());
                     }
 
                   fe_values.reinit (cell, face + cell->active_fe_index ()
@@ -6166,10 +6151,8 @@ namespace VectorTools
                   // check this.
                   if (dynamic_cast<const FESystem<dim>*> (&cell->get_fe ()) == nullptr)
                     {
-                      typedef FiniteElement<dim> FEL;
-
                       AssertThrow (dynamic_cast<const FE_RaviartThomas<dim>*> (&cell->get_fe ()) != nullptr,
-                                   typename FEL::ExcInterpolationNotImplemented ());
+                                   typename FiniteElement<dim>::ExcInterpolationNotImplemented ());
                     }
 
                   fe_values.reinit (cell, face + cell->active_fe_index ()
@@ -6213,10 +6196,8 @@ namespace VectorTools
                   // check this.
                   if (dynamic_cast<const FESystem<dim>*> (&cell->get_fe ()) == nullptr)
                     {
-                      typedef FiniteElement<dim> FEL;
-
                       AssertThrow (dynamic_cast<const FE_RaviartThomas<dim>*> (&cell->get_fe ()) != nullptr,
-                                   typename FEL::ExcInterpolationNotImplemented ());
+                                   typename FiniteElement<dim>::ExcInterpolationNotImplemented ());
                     }
 
                   fe_values.reinit (cell, face + cell->active_fe_index ()

--- a/source/fe/fe.cc
+++ b/source/fe/fe.cc
@@ -849,10 +849,9 @@ get_interpolation_matrix (const FiniteElement<dim,spacedim> &,
   // by default, no interpolation
   // implemented. so throw exception,
   // as documentation says
-  typedef FiniteElement<dim,spacedim> FEE;
   AssertThrow (false,
-               typename FEE::
-               ExcInterpolationNotImplemented());
+               (typename FiniteElement<dim,spacedim>::
+                ExcInterpolationNotImplemented()));
 }
 
 
@@ -866,10 +865,9 @@ get_face_interpolation_matrix (const FiniteElement<dim,spacedim> &,
   // by default, no interpolation
   // implemented. so throw exception,
   // as documentation says
-  typedef    FiniteElement<dim,spacedim> FEE;
   AssertThrow (false,
-               typename FEE::
-               ExcInterpolationNotImplemented());
+               (typename FiniteElement<dim,spacedim>::
+                ExcInterpolationNotImplemented()));
 }
 
 
@@ -884,9 +882,9 @@ get_subface_interpolation_matrix (const FiniteElement<dim,spacedim> &,
   // by default, no interpolation
   // implemented. so throw exception,
   // as documentation says
-  typedef    FiniteElement<dim,spacedim> FEE;
   AssertThrow (false,
-               typename FEE::ExcInterpolationNotImplemented());
+               (typename FiniteElement<dim,spacedim>::
+                ExcInterpolationNotImplemented()));
 }
 
 

--- a/source/fe/fe_bernstein.cc
+++ b/source/fe/fe_bernstein.cc
@@ -53,9 +53,9 @@ get_interpolation_matrix (const FiniteElement<dim,spacedim> &,
                           FullMatrix<double> &) const
 {
   // no interpolation possible. throw exception, as documentation says
-  typedef FiniteElement<dim,spacedim> FEE;
   AssertThrow (false,
-               typename FEE::ExcInterpolationNotImplemented());
+               (typename FiniteElement<dim,spacedim>::
+                ExcInterpolationNotImplemented()));
 }
 
 
@@ -65,9 +65,8 @@ const FullMatrix<double> &
 FE_Bernstein<dim,spacedim>::get_restriction_matrix (const unsigned int,
                                                     const RefinementCase<dim> &) const
 {
-  typedef FiniteElement<dim,spacedim> FEE;
   AssertThrow (false,
-               typename FEE::ExcProjectionVoid());
+               (typename FiniteElement<dim,spacedim>::ExcProjectionVoid()));
   // return dummy, nothing will happen because the base class FE_Q_Base
   // implements lazy evaluation of those matrices
   return this->restriction[0][0];
@@ -80,9 +79,8 @@ const FullMatrix<double> &
 FE_Bernstein<dim,spacedim>::get_prolongation_matrix (const unsigned int,
                                                      const RefinementCase<dim> &) const
 {
-  typedef FiniteElement<dim,spacedim> FEE;
   AssertThrow (false,
-               typename FEE::ExcEmbeddingVoid());
+               (typename FiniteElement<dim,spacedim>::ExcEmbeddingVoid()));
   // return dummy, nothing will happen because the base class FE_Q_Base
   // implements lazy evaluation of those matrices
   return this->prolongation[0][0];

--- a/source/fe/fe_dgp_nonparametric.cc
+++ b/source/fe/fe_dgp_nonparametric.cc
@@ -431,12 +431,11 @@ get_face_interpolation_matrix (const FiniteElement<dim,spacedim> &x_source_fe,
   // is necessarily empty -- i.e. there isn't
   // much we need to do here.
   (void)interpolation_matrix;
-  typedef              FiniteElement<dim,spacedim> FEE;
   AssertThrow ((x_source_fe.get_name().find ("FE_DGPNonparametric<") == 0)
                ||
                (dynamic_cast<const FE_DGPNonparametric<dim,spacedim>*>(&x_source_fe) != nullptr),
-               typename FEE::
-               ExcInterpolationNotImplemented());
+               (typename FiniteElement<dim,spacedim>::
+                ExcInterpolationNotImplemented()));
 
   Assert (interpolation_matrix.m() == 0,
           ExcDimensionMismatch (interpolation_matrix.m(),
@@ -462,12 +461,10 @@ get_subface_interpolation_matrix (const FiniteElement<dim,spacedim> &x_source_fe
   // is necessarily empty -- i.e. there isn't
   // much we need to do here.
   (void)interpolation_matrix;
-  typedef              FiniteElement<dim,spacedim> FEE;
   AssertThrow ((x_source_fe.get_name().find ("FE_DGPNonparametric<") == 0)
                ||
                (dynamic_cast<const FE_DGPNonparametric<dim,spacedim>*>(&x_source_fe) != nullptr),
-               typename FEE::
-               ExcInterpolationNotImplemented());
+               (typename FiniteElement<dim,spacedim>::ExcInterpolationNotImplemented()));
 
   Assert (interpolation_matrix.m() == 0,
           ExcDimensionMismatch (interpolation_matrix.m(),

--- a/source/fe/fe_enriched.cc
+++ b/source/fe/fe_enriched.cc
@@ -790,8 +790,8 @@ get_face_interpolation_matrix (const FiniteElement<dim,spacedim> &source,
     }
   else
     {
-      typedef FiniteElement<dim,spacedim> FEL;
-      AssertThrow(false,typename FEL::ExcInterpolationNotImplemented());
+      AssertThrow(false, (typename FiniteElement<dim,spacedim>::
+                          ExcInterpolationNotImplemented()));
     }
 }
 
@@ -812,8 +812,8 @@ get_subface_interpolation_matrix (const FiniteElement<dim,spacedim> &source,
     }
   else
     {
-      typedef FiniteElement<dim,spacedim> FEL;
-      AssertThrow(false,typename FEL::ExcInterpolationNotImplemented());
+      AssertThrow(false,(typename FiniteElement<dim,spacedim>::
+                         ExcInterpolationNotImplemented()));
     }
 }
 

--- a/source/fe/fe_nedelec.cc
+++ b/source/fe/fe_nedelec.cc
@@ -2471,12 +2471,10 @@ const
   // this is only implemented, if the
   // source FE is also a
   // Nedelec element
-  typedef FE_Nedelec<dim> FEN;
-  typedef FiniteElement<dim> FEL;
-
   AssertThrow ((source.get_name ().find ("FE_Nedelec<") == 0) ||
-               (dynamic_cast<const FEN *> (&source) != nullptr),
-               typename FEL::ExcInterpolationNotImplemented());
+               (dynamic_cast<const FE_Nedelec<dim> *> (&source) != nullptr),
+               (typename FiniteElement<dim>::
+                ExcInterpolationNotImplemented()));
   Assert (interpolation_matrix.m () == source.dofs_per_face,
           ExcDimensionMismatch (interpolation_matrix.m (),
                                 source.dofs_per_face));
@@ -2501,7 +2499,8 @@ const
   // hp procedures, which use this
   // method.
   Assert (this->dofs_per_face <= source_fe.dofs_per_face,
-          typename FEL::ExcInterpolationNotImplemented ());
+          (typename FiniteElement<dim>::
+           ExcInterpolationNotImplemented ()));
   interpolation_matrix = 0;
 
   // On lines we can just identify
@@ -2579,12 +2578,10 @@ FE_Nedelec<dim>::get_subface_interpolation_matrix(
   // this is only implemented, if the
   // source FE is also a
   // Nedelec element
-  typedef FE_Nedelec<dim> FEN;
-  typedef FiniteElement<dim> FEL;
-
   AssertThrow ((source.get_name ().find ("FE_Nedelec<") == 0) ||
-               (dynamic_cast<const FEN *> (&source) != nullptr),
-               typename FEL::ExcInterpolationNotImplemented ());
+               (dynamic_cast<const FE_Nedelec<dim> *> (&source) != nullptr),
+               typename FiniteElement<dim>::
+               ExcInterpolationNotImplemented ());
   Assert (interpolation_matrix.m () == source.dofs_per_face,
           ExcDimensionMismatch (interpolation_matrix.m (),
                                 source.dofs_per_face));
@@ -2609,7 +2606,8 @@ FE_Nedelec<dim>::get_subface_interpolation_matrix(
   // hp procedures, which use this
   // method.
   Assert (this->dofs_per_face <= source_fe.dofs_per_face,
-          typename FEL::ExcInterpolationNotImplemented ());
+          (typename FiniteElement<dim>::
+           ExcInterpolationNotImplemented ()));
   interpolation_matrix = 0.0;
   // Perform projection-based interpolation
   // as usual.

--- a/source/fe/fe_poly_tensor.cc
+++ b/source/fe/fe_poly_tensor.cc
@@ -171,8 +171,7 @@ FE_PolyTensor<PolynomialType,dim,spacedim>::shape_value
 (const unsigned int, const Point<dim> &) const
 
 {
-  typedef    FiniteElement<dim,spacedim> FEE;
-  Assert(false, typename FEE::ExcFENotPrimitive());
+  Assert(false, (typename FiniteElement<dim,spacedim>::ExcFENotPrimitive()));
   return 0.;
 }
 
@@ -216,8 +215,7 @@ Tensor<1,dim>
 FE_PolyTensor<PolynomialType,dim,spacedim>::shape_grad (const unsigned int,
                                                         const Point<dim> &) const
 {
-  typedef    FiniteElement<dim,spacedim> FEE;
-  Assert(false, typename FEE::ExcFENotPrimitive());
+  Assert(false, (typename FiniteElement<dim,spacedim>::ExcFENotPrimitive()));
   return Tensor<1,dim>();
 }
 
@@ -263,8 +261,7 @@ FE_PolyTensor<PolynomialType,dim,spacedim>::shape_grad_grad
 (const unsigned int,
  const Point<dim> &) const
 {
-  typedef    FiniteElement<dim,spacedim> FEE;
-  Assert(false, typename FEE::ExcFENotPrimitive());
+  Assert(false, (typename FiniteElement<dim,spacedim>::ExcFENotPrimitive()));
   return Tensor<2,dim>();
 }
 

--- a/source/fe/fe_q_bubbles.cc
+++ b/source/fe/fe_q_bubbles.cc
@@ -371,14 +371,13 @@ get_interpolation_matrix (const FiniteElement<dim,spacedim> &x_source_fe,
   // However, for SolutionTransfer to work we need to provide an implementation
   // for the case that the x_source_fe is identical to this FE
   typedef FE_Q_Bubbles<dim,spacedim> FEQBUBBLES;
-  typedef FiniteElement<dim,spacedim> FEL;
 
   AssertThrow ((x_source_fe.get_name().find ("FE_Q_Bubbles<") == 0)
                ||
                (dynamic_cast<const FEQBUBBLES *>(&x_source_fe) != nullptr)
                ,
-               typename FEL::
-               ExcInterpolationNotImplemented());
+               (typename FiniteElement<dim,spacedim>::
+                ExcInterpolationNotImplemented()));
   Assert (interpolation_matrix.m() == this->dofs_per_cell,
           ExcDimensionMismatch (interpolation_matrix.m(),
                                 this->dofs_per_cell));
@@ -392,8 +391,9 @@ get_interpolation_matrix (const FiniteElement<dim,spacedim> &x_source_fe,
       interpolation_matrix.set(i,i,1.);
   //else we need to do more...
   else
-    Assert(false, typename FEL::ExcInterpolationNotImplemented())
-  }
+    Assert(false, (typename FiniteElement<dim,spacedim>::
+                   ExcInterpolationNotImplemented()));
+}
 
 
 

--- a/source/fe/fe_q_dg0.cc
+++ b/source/fe/fe_q_dg0.cc
@@ -210,13 +210,12 @@ get_interpolation_matrix (const FiniteElement<dim,spacedim> &x_source_fe,
 {
   // this is only implemented, if the source FE is also a Q_DG0 element
   typedef FE_Q_DG0<dim,spacedim> FEQDG0;
-  typedef FiniteElement<dim,spacedim> FEL;
 
   AssertThrow ((x_source_fe.get_name().find ("FE_Q_DG0<") == 0)
                ||
                (dynamic_cast<const FEQDG0 *>(&x_source_fe) != nullptr),
-               typename FEL::
-               ExcInterpolationNotImplemented());
+               (typename FiniteElement<dim,spacedim>::
+                ExcInterpolationNotImplemented()));
 
   Assert (interpolation_matrix.m() == this->dofs_per_cell,
           ExcDimensionMismatch (interpolation_matrix.m(),

--- a/source/fe/fe_q_hierarchical.cc
+++ b/source/fe/fe_q_hierarchical.cc
@@ -828,12 +828,10 @@ get_face_interpolation_matrix (const FiniteElement<dim> &x_source_fe,
   // source FE is also a
   // Q_Hierarchical element
   typedef FE_Q_Hierarchical<dim> FEQHierarchical;
-  typedef FiniteElement<dim> FEL;
   AssertThrow ((x_source_fe.get_name().find ("FE_Q_Hierarchical<") == 0)
                ||
                (dynamic_cast<const FEQHierarchical *>(&x_source_fe) != nullptr),
-               typename FEL::
-               ExcInterpolationNotImplemented());
+               (typename FiniteElement<dim>::ExcInterpolationNotImplemented()));
 
   Assert (interpolation_matrix.n() == this->dofs_per_face,
           ExcDimensionMismatch (interpolation_matrix.n(),
@@ -860,8 +858,8 @@ get_face_interpolation_matrix (const FiniteElement<dim> &x_source_fe,
   // hp procedures, which use this
   // method.
   Assert (this->dofs_per_face <= source_fe.dofs_per_face,
-          typename FEL::
-          ExcInterpolationNotImplemented ());
+          (typename FiniteElement<dim>::
+           ExcInterpolationNotImplemented ()));
   interpolation_matrix = 0;
 
   switch (dim)
@@ -916,12 +914,10 @@ get_subface_interpolation_matrix (const FiniteElement<dim> &x_source_fe,
   // source FE is also a
   // Q_Hierarchical element
   typedef FE_Q_Hierarchical<dim> FEQHierarchical;
-  typedef FiniteElement<dim> FEL;
   AssertThrow ((x_source_fe.get_name().find ("FE_Q_Hierarchical<") == 0)
                ||
                (dynamic_cast<const FEQHierarchical *>(&x_source_fe) != nullptr),
-               typename FEL::
-               ExcInterpolationNotImplemented());
+               (typename FiniteElement<dim>::ExcInterpolationNotImplemented()));
 
   Assert (interpolation_matrix.n() == this->dofs_per_face,
           ExcDimensionMismatch (interpolation_matrix.n(),
@@ -947,8 +943,8 @@ get_subface_interpolation_matrix (const FiniteElement<dim> &x_source_fe,
   // hp procedures, which use this
   // method.
   Assert (this->dofs_per_face <= source_fe.dofs_per_face,
-          typename FEL::
-          ExcInterpolationNotImplemented ());
+          (typename FiniteElement<dim>::
+           ExcInterpolationNotImplemented ()));
 
   switch (dim)
     {

--- a/source/fe/fe_system.cc
+++ b/source/fe/fe_system.cc
@@ -615,12 +615,11 @@ FESystem<dim,spacedim>::get_interpolation_matrix (
   //
   // condition 1: the other element must also be a system element
 
-  typedef FiniteElement<dim,spacedim> FEL;
   AssertThrow ((x_source_fe.get_name().find ("FESystem<") == 0)
                ||
                (dynamic_cast<const FESystem<dim,spacedim>*>(&x_source_fe) != nullptr),
-               typename FEL::
-               ExcInterpolationNotImplemented());
+               (typename FiniteElement<dim,spacedim>::
+                ExcInterpolationNotImplemented()));
 
   // ok, source is a system element, so we may be able to do the work
   const FESystem<dim,spacedim> &source_fe
@@ -628,15 +627,15 @@ FESystem<dim,spacedim>::get_interpolation_matrix (
 
   // condition 2: same number of basis elements
   AssertThrow (this->n_base_elements() == source_fe.n_base_elements(),
-               typename FEL::
-               ExcInterpolationNotImplemented());
+               (typename FiniteElement<dim,spacedim>::
+                ExcInterpolationNotImplemented()));
 
   // condition 3: same number of basis elements
   for (unsigned int i=0; i< this->n_base_elements(); ++i)
     AssertThrow (this->element_multiplicity(i) ==
                  source_fe.element_multiplicity(i),
-                 typename FEL::
-                 ExcInterpolationNotImplemented());
+                 (typename FiniteElement<dim,spacedim>::
+                  ExcInterpolationNotImplemented()));
 
   // ok, so let's try whether it works:
 
@@ -1745,12 +1744,11 @@ FESystem<dim,spacedim>::
 get_face_interpolation_matrix (const FiniteElement<dim,spacedim> &x_source_fe,
                                FullMatrix<double>       &interpolation_matrix) const
 {
-  typedef FiniteElement<dim,spacedim> FEL;
   AssertThrow ((x_source_fe.get_name().find ("FE_System<") == 0)
                ||
                (dynamic_cast<const FESystem<dim,spacedim>*>(&x_source_fe) != nullptr),
-               typename FEL::
-               ExcInterpolationNotImplemented());
+               (typename FiniteElement<dim,spacedim>::
+                ExcInterpolationNotImplemented()));
 
   Assert (interpolation_matrix.n() == this->dofs_per_face,
           ExcDimensionMismatch (interpolation_matrix.n(),
@@ -1856,12 +1854,11 @@ get_subface_interpolation_matrix (const FiniteElement<dim,spacedim> &x_source_fe
                                   const unsigned int        subface,
                                   FullMatrix<double>       &interpolation_matrix) const
 {
-  typedef FiniteElement<dim,spacedim> FEL;
   AssertThrow ((x_source_fe.get_name().find ("FE_System<") == 0)
                ||
                (dynamic_cast<const FESystem<dim,spacedim>*>(&x_source_fe) != nullptr),
-               typename FEL::
-               ExcInterpolationNotImplemented());
+               (typename FiniteElement<dim,spacedim>::
+                ExcInterpolationNotImplemented()));
 
   Assert (interpolation_matrix.n() == this->dofs_per_face,
           ExcDimensionMismatch (interpolation_matrix.n(),
@@ -2172,10 +2169,9 @@ FESystem<dim,spacedim>::unit_support_point (const unsigned int index) const
 {
   Assert (index < this->dofs_per_cell,
           ExcIndexRange (index, 0, this->dofs_per_cell));
-  typedef FiniteElement<dim,spacedim> FEL;
   Assert ((this->unit_support_points.size() == this->dofs_per_cell) ||
           (this->unit_support_points.size() == 0),
-          typename FEL::ExcFEHasNoSupportPoints ());
+          (typename FiniteElement<dim,spacedim>::ExcFEHasNoSupportPoints ()));
 
   // let's see whether we have the information pre-computed
   if (this->unit_support_points.size() != 0)
@@ -2195,10 +2191,9 @@ FESystem<dim,spacedim>::unit_face_support_point (const unsigned int index) const
 {
   Assert (index < this->dofs_per_face,
           ExcIndexRange (index, 0, this->dofs_per_face));
-  typedef  FiniteElement<dim,spacedim> FEL;
   Assert ((this->unit_face_support_points.size() == this->dofs_per_face) ||
           (this->unit_face_support_points.size() == 0),
-          typename FEL::ExcFEHasNoSupportPoints ());
+          (typename FiniteElement<dim,spacedim>::ExcFEHasNoSupportPoints ()));
 
   // let's see whether we have the information pre-computed
   if (this->unit_face_support_points.size() != 0)

